### PR TITLE
[Feature] support create iceberg database

### PR DESCRIPTION
**Pull request by stephen-shelby**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #
Support create iceberg database in the iceberg catalog.
- create database iceberg_db 
- create database iceberg_catalog.iceberg_db
- create database iceberg_catalog.iceberg_db properties ("location" = "hdfs://hadoop:9000/user/hive/warehouse/iceberg_db.db")

location in properties mean iceberg database location. if not set, it will use the default location in the user meta service.

https://github.com/StarRocks/starrocks/issues/21502
## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Review by Astralidea — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/iceberg/hive/IcebergHiveCatalog.java [hunk 1]
```diff
@@ -200,7 +251,11 @@ public void setConf(Configuration conf) {
         this.conf = conf;
     }
 
-    @Override
+    // for unit test
+    public ClientPool<IMetaStoreClient, TException> getClients() {
+        return clients;
+    }
+
     public String toString() {
```
Comment (zombee0) [inline fe/fe-core/src/main/java/org/apache/iceberg/hive/IcebergHiveCatalog.java:259]: Override

**Review by Youngwb — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/iceberg/hive/IcebergHiveCatalog.java [hunk 2]
```diff
@@ -174,6 +180,51 @@ public Database getDB(String dbName) throws InterruptedException, TException {
...
+        Database database = new Database(CONNECTOR_ID_GENERATOR.getNextId().asInt(), dbName);
+        properties = properties == null ? new HashMap<>() : properties;
+        for (Map.Entry<String, String> entry : properties.entrySet()) {
+            String key = entry.getKey();
+            String value = entry.getValue();
+            if (key.equalsIgnoreCase(LOCATION_PROPERTY)) {
+                try {
+                    URI uri = new Path(value).toUri();
+                    FileSystem fileSystem = FileSystem.get(uri, new Configuration());
+                    fileSystem.exists(new Path(value));
```
Comment (zombee0) [inline fe/fe-core/src/main/java/org/apache/iceberg/hive/IcebergHiveCatalog.java:194]: check the path exists? why not deal the return value?

**Review by zombee0 — APPROVED** [review]
Some comments, you can deal in later pr.

File: fe/fe-core/src/main/java/org/apache/iceberg/hive/IcebergHiveCatalog.java (hunk 2 above)
Comment (stephen-shelby) [reply fe/fe-core/src/main/java/org/apache/iceberg/hive/IcebergHiveCatalog.java:194]: It's mainly to check if the path could access in filesystem

File: fe/fe-core/src/test/java/com/starrocks/connector/iceberg/IcebergMetadataTest.java [hunk 3]
```diff
@@ -104,7 +111,9 @@ public void testGetTable(@Mocked IcebergHiveCatalog icebergHiveCatalog,
 
         IcebergMetadata metadata = new IcebergMetadata(CATALOG_NAME, icebergHiveCatalog);
         Table expectResult = new Table(100000000, "tbl", ICEBERG, new ArrayList<>());
```
[the remark this answers is not in the document]
Comment (stephen-shelby) [reply fe/fe-core/src/test/java/com/starrocks/connector/iceberg/IcebergMetadataTest.java:113]: I will update it in next pr

File: fe/fe-core/src/main/java/org/apache/iceberg/hive/IcebergHiveCatalog.java (hunk 1 above)
Comment (stephen-shelby) [reply fe/fe-core/src/main/java/org/apache/iceberg/hive/IcebergHiveCatalog.java:259]: ok

**Pull request merged**